supertokens/userIdMapping.go

```diff
@@ -0,0 +1,217 @@
+package supertokens
+
+import (
+	"errors"
+)
+
+var userIdTypeDefault = "ANY"
+
+type CreateUserIdMappingResult struct {
+	OK                              *struct{}
+	UnknownSupertokensUserIdError   *struct{}
+	UserIdMappingAlreadyExistsError *struct {
+		DoesSuperTokensUserIdExist bool
+		DoesExternalUserIdExist    bool
+	}
+}
+
+func CreateUserIdMapping(supertokensUserId string, externalUserId string, externalUserIdInfo *string, force bool) (CreateUserIdMappingResult, error) {
+	querier, err := GetNewQuerierInstanceOrThrowError("")
+	if err != nil {
+		return CreateUserIdMappingResult{}, err
+	}
+	cdiVersion, err := querier.GetQuerierAPIVersion()
+	if err != nil {
+		return CreateUserIdMappingResult{}, err
+	}
+	if maxVersion(cdiVersion, "2.15") != cdiVersion {
+		return CreateUserIdMappingResult{}, errors.New("Please upgrade the SuperTokens core to >= 3.15.0")
+	}
+
+	data := map[string]interface{}{
+		"superTokensUserId": supertokensUserId,
+		"externalUserId":    externalUserId,
+		"force":             force,
+	}
+	if externalUserIdInfo != nil {
+		data["externalUserIdInfo"] = *externalUserIdInfo
+	}
+	resp, err := querier.SendPostRequest("/recipe/userid/map", data)
+	if err != nil {
+		return CreateUserIdMappingResult{}, err
+	}
+	if resp["status"] == "OK" {
+		return CreateUserIdMappingResult{
+			OK: &struct{}{},
+		}, nil
+	} else if resp["status"] == "UNKNOWN_SUPERTOKENS_USER_ID_ERROR" {
+		return CreateUserIdMappingResult{
+			UnknownSupertokensUserIdError: &struct{}{},
+		}, nil
+	} else {
+		return CreateUserIdMappingResult{
+			UserIdMappingAlreadyExistsError: &struct {
+				DoesSuperTokensUserIdExist bool
+				DoesExternalUserIdExist    bool
+			}{
+				DoesSuperTokensUserIdExist: resp["doesSuperTokensUserIdExist"].(bool),
+				DoesExternalUserIdExist:    resp["doesExternalUserIdExist"].(bool),
+			},
+		}, nil
+	}
+}
+
+type GetUserIdMappingResult struct {
+	OK *struct {
+		SupertokensUserId  string
+		ExternalUserId     string
+		ExternalUserIdInfo *string
+	}
+	UnknownMappingError *struct{}
+}
+
+func GetUserIdMapping(userId string, userIdType *string) (GetUserIdMappingResult, error) {
```

Comment:
Make userIdType to enum since it only has certain values.

Reply:
Done